### ISSUE TITLE
Display nice tracebacks when deploy fails

### ISSUE THREAD

**Issue by pablohoffman** [scrapinghub/shub#167]
When deployment fails, it would be nice to display a nice traback instead of a blob of JSON. After all, the traceback is there, just poorly formatted.

So, instead of showing this:

```
Error: Deploy failed (400):
{"status": "badrequest", "message": "ERROR:root:Script initialization failed\nTraceback (most recent call last):\n  File \"/usr/lib/python2.7/dist-packages/sh_scrapy/crawl.py\", line 127, in _run_usercode\n    _run(args, settings)\n  File \"/usr/lib/python2.7/dist-packages/sh_scrapy/crawl.py\", line 87, in _run\n    _run_scrapy(args, settings)\n  File \"/usr/lib/python2.7/dist-packages/sh_scrapy/crawl.py\", line 95, in _run_scrapy\n    execute(settings=settings)\n  File \"/usr/lib/pymodules/python2.7/scrapy/cmdline.py\", line 142, in execute\n    cmd.crawler_process = CrawlerProcess(settings)\n  File \"/usr/lib/pymodules/python2.7/scrapy/crawler.py\", line 209, in __init__\n    super(CrawlerProcess, self).__init__(settings)\n  File \"/usr/lib/pymodules/python2.7/scrapy/crawler.py\", line 115, in __init__\n    self.spider_loader = _get_spider_loader(settings)\n  File \"/usr/lib/pymodules/python2.7/scrapy/crawler.py\", line 296, in _get_spider_loader\n    return loader_cls.from_settings(settings.frozencopy())\n  File \"/usr/lib/pymodules/python2.7/scrapy/spiderloader.py\", line "}
```

We should show something like:

```
Error: Deploy failed (400)
Traceback (most recent call last):
  File "/usr/local/bin/scrapy", line 11, in <module>
    sys.exit(execute())
  File "/usr/local/lib/python2.7/dist-packages/scrapy/cmdline.py", line 142, in execute
    cmd.crawler_process = CrawlerProcess(settings)
  File "/usr/local/lib/python2.7/dist-packages/scrapy/crawler.py", line 209, in __init__
    super(CrawlerProcess, self).__init__(settings)
  File "/usr/local/lib/python2.7/dist-packages/scrapy/crawler.py", line 115, in __init__
    self.spider_loader = _get_spider_loader(settings)
  File "/usr/local/lib/python2.7/dist-packages/scrapy/crawler.py", line 296, in _get_spider_loader
    return loader_cls.from_settings(settings.frozencopy())
  File "/usr/local/lib/python2.7/dist-packages/scrapy/spiderloader.py", line 30, in from_settings
    return cls(settings)
  File "/usr/local/lib/python2.7/dist-packages/scrapy/spiderloader.py", line 21, in __init__
    for module in walk_modules(name):
  File "/usr/local/lib/python2.7/dist-packages/scrapy/utils/misc.py", line 71, in walk_modules
    submod = import_module(fullpath)
  File "/usr/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/app/__main__.egg/xxxxxx/spiders/glg.py", line 21
    for category in categories):
```

I consider this a pretty important enhancement for SC user experience.

**Comment by jvsteiner** [comment]
I am also unable to debug a spider deployment, because of this problem.  The JSON response is incomplete, making it impossible for me to tell which line of my code is causing a problem, since the bottom of the stacktrace is missing.

**Comment by pablohoffman** [comment]
@jvsteiner the incomplete JSON response problem is fixed now (can you try it and confirm?), the rendering improvements will come shortly to `shub`.

**Comment by jvsteiner** [comment]
Hi Pablo - the stacktrace is now complete - many thanks!

**Comment by pablohoffman** [comment]
You're welcome @jvsteiner, there is already a pull request for improving traceback rendering (#170) which should be merged soon and released in shub 2.0.2
